Mathlib/Algebra/Polynomial/ofFn.lean

```diff
@@ -0,0 +1,123 @@
+/-
+Copyright (c) 2025 Fabrizio Barroero. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Fabrizio Barroero
+-/
+import Mathlib.Algebra.BigOperators.Fin
+import Mathlib.Algebra.Polynomial.Degree.Lemmas
+import Mathlib.Data.List.ToFinsupp
+/-!
+# `Polynomial.ofFn` and `Polynomial.toFn`
+
+In this file we introduce `ofFn` and `toFn`, two functions that associate a polynomial to the vector
+of its coefficients and vice versa. We prove some basic APIs for these functions.
+
+## Main definitions
+
+- `Polynomial.toFn n` associates to a polynomial the vector of its first `n` coefficients.
+- `Polynomial.ofFn n` associates to a vector of length `n` the polynomial that has the entries of
+  the vector as coefficients.
+-/
+
+namespace Polynomial
+
+section toFn
+
+variable {R : Type*} [Semiring R]
+
+/-- `toFn n f` is the vector of the first `n` coefficients of the polynomial `f`. -/
+def toFn (n : ℕ) : R[X] →ₗ[R] Fin n → R where
+  toFun p := fun i ↦ p.coeff i
+  map_add' x y := by
+    simp only [coeff_add]
+    rfl
+  map_smul' x p := by
+    simp only [coeff_smul]
+    rfl
```

Comment:
You should be able to build this by composing `Polynomial.lcoeff` with some trivial reindexing map that surely exists (and perhaps `apply?` can find.